discretize/mixins/vtkInterface.py

```diff
@@ -141,6 +112,60 @@ def __createStructGrid(ptsMat, dims, models=None):
         # Assign the model('s) to the object
         return vtkInterface.__assignCellData(output, models=models)
 
+    def __getRotatedNodes(mesh):
+        """A helper to get the nodes of a mesh rotated by specified axes"""
+        nodes = mesh.gridN
+        if mesh.dim == 1:
+            nodes = np.c_[mesh.gridN, np.zeros((mesh.nN, 2))]
+        elif mesh.dim == 2:
+            nodes = np.c_[mesh.gridN, np.zeros((mesh.nN, 1))]
+        # Now garuntee nodes are correct
+        if nodes.shape != (mesh.nN, 3):
+            raise RuntimeError('Nodes of the grid are improperly defined.')
+        # Rotate the points based on the axis orientations
+        mesh._validate_orientation()
+        return np.dot(nodes, mesh.rotMtx)
+
+    def __tensorMeshToVTK(mesh, models=None):
```

Comment:
should this be name mangled  with two `_` or just private?

Reply:
My reasoning for name mangling with two `_`s was that the `BaseMesh` class inherits the `vtkInterface` class and thus all of these functions would be available to any class inheriting the `BaseMesh` if they were private. By name mangling and providing a wrapper in the `toVTK` and `writeVTK` methods, we prevent users from accidentally calling the wrong conversion method.

If we made them private, then I'd be concerned about the ability to do this:
```py
>>> tens = discretize.TensorMesh([16, 16, 16])
>>> # not okay call:
>>> tens._treeMeshToVTK()
Exception: 'some sort of crazy error message'
```

However, there are a few name mangled methods in there that would make sense for any mesh type to have available:

- `__assignCellData`
- `__getRotatedNodes`